Add support for draft documentation pages (#1613)
withastro/starlight: docs/src/content/docs/guides/pages.mdx

- The [`slug`](/reference/frontmatter/#slug) property is not supported and is automatically set based on the custom page’s URL.
- The [`editUrl`](/reference/frontmatter/#editurl) option requires a URL to display an edit link.
- The [`sidebar`](/reference/frontmatter/#sidebar) frontmatter property for customizing how the page appears in [autogenerated link groups](/reference/configuration/#sidebar) is not available. Pages using the `<StarlightPage />` component are not part of a collection and cannot be added to an autogenerated sidebar group.

> - The [`draft`](/reference/frontmatter/#draft) option only displays a [notice](/reference/overrides/#draftcontentnotice) that the page is a draft but does not automatically exclude it from production builds.

Reviewer: Is this correct or should it read like this:

```suggestion
- The [`draft`](/reference/frontmatter/#draft) option displays a [notice](/reference/overrides/#draftcontentnotice) that the page is a draft and will not be included in production builds.
```

Author: This is actually correct in this specific section as we are talking about custom pages automatically injected by a plugin for example so it would be up to the plugin to make sure the page is not injected when not needed.

Altho, this may also be poorly worded as I'm the one who wrote it so may need some tweaking ^^
